processes/loc.py

```diff
@@ -37,27 +39,49 @@ def __init__(self, *args):
         self.createS3Client()
 
     def runProcess(self):
+        if self.process == 'weekly':
+            self.importLOCRecords()
+        elif self.process == 'complete':
+            self.importLOCRecords(fullImport=True)
+        elif self.process == 'custom':
+            self.importLOCRecords(startTimestamp=self.ingestPeriod)
+
+        self.saveRecords()
+        self.commitChanges()
+    
+
+    def importLOCRecords(self, fullImport=False, startTimestamp=None):
+
+        if not fullImport:
+            if not startTimestamp:
+                startTimestamp = datetime.utcnow() - timedelta(days=7)
+            else:
+                startTimestamp = datetime.strptime(startTimestamp, '%Y-%m-%dT%H:%M:%S')
```

Comment:
I have a small suggestion to simplify some of these flags / logic, but lmk what you think / if I'm missing something!

It seems like there are 3 cases we actually expect to see:

  1.   fullImport=True and startTimestamp=None
  2.  fullImport=False and startTimestamp=None (which means weekly)
  3. fullImport=False and startTimestamp is set

I think we can cover this same logic *without* the fullImport flag, by passing in a timestamp to `importLOCRecords` even for the weekly ingest.  So move the timestamp logic for a weekly ingest up to the `if self.process = 'weekly'` block in `runProcess`.

Then you can get rid of the if block here at the start of `importLOCRecords`. It is also more intuitive as a reader (at least to me) that when you call `importLOCRecords()` with no args, it's a full import, and if you call it like `importLOCRecords(timestamp)` it always uses the start timestamp.  Right now, it's a little confusing that if you call `self.importLOCRecords()` with no timestamps, it's actually importing a week's worth of records implicitly.

Does that make sense? Or am I missing something here?

Reply:
I agree with your assessment and I can see how it's set up now can be confusing. I'll incorporate the timeStamp variable only and get rid of the fullImport variable.